In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
# read the first sheet of the excel file
df = pd.read_excel('../datasets/aku_prin_v2.0.xlsx', sheet_name=0)

In [4]:
df.columns

Index(['patient', 'sex', 'birth', 'creatinine_urines mg/dl', 'HGA_u mg/dl',
       'HGA_U mg/24h', 'bqa', 'urate', 'uric_acid_urine', 'tyr_urines',
       'hypoxanthine_urines', 'xanthine_urines', 'Protein change allele 1 ',
       'DNA change allele 1', 'Exon/intron allele 1',
       'Protein change allele 2', 'DNA change allele 2', 'ex/in allele 2',
       'SAA (ug/mL) ', 'chitotriosidase', 'HGA_serum', 'CATD (ng/mL)',
       'IL-6 (pg/mL)', 'IL-1beta (pg/mL)', 'IL-1ra (pg/mL)', 'TNFalfa (pg/mL)',
       'CRP (mg/dL)', 'CRP (æg/mL) ELISA', 'MMP3 (ng/mL)', 'aopp',
       'glucose(mg/dL)', 'creatinine (mg/dL)', 'cholesterol (mg/dL)',
       'triglycerides  (mg/dL)', 'hdl_cholesterol (mg/dL)',
       'ldl_cholesterol (mg/dL)', 'alkaline_phosphatase (UI/L)',
       'cystatin_C (mg/dL)', 'smoker/cigarettes_a_day', 'alcohol_units_weekly',
       'bmi', 'physical_health_score', 'mental_health_score',
       'AKUSSI_jointpain', 'AKUSSI_spinalpain', 'KOOSpain', 'KOOSsymptoms',
       'KOOSdai

In [5]:
len(df.columns)

67

In [6]:
# get the index of Protein change allele 1 column
df.columns.get_loc('Protein change allele 1 ') +6


18

In [7]:
# get the index of physical_health_score column
df.columns.get_loc('physical_health_score') + 11

52

In [8]:
# create a new dataframe with the columns of interest, from index 12 to 18 and 41 to 52
df2 = df.iloc[:, list(range(12, 18)) + list(range(41, 52))]
df2.head()


Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
0                    G161R            c.481G>A                  ex8   
1                    G161R            c.481G>A                  ex8   
2                    G161R            c.481G>A                  ex8   
3      Y6_G29del ivs1-1G>A           c.16-1G>A                  in1   
4                    G161R            c.481G>A                  ex8   

  Protein change allele 2 DNA change allele 2 ex/in allele 2  \
0             His371Profs          c.1111dupC           ex13   
1                   G161R            c.481G>A            ex8   
2                   M368V           c.1102A>G           ex13   
3     Y6_G29del ivs1-1G>A           c.16-1G>A            in1   
4                   G161R            c.481G>A            ex8   

   physical_health_score  mental_health_score  AKUSSI_jointpain  \
0                   26.0                 43.0              57.0   
1                   53.0                 51.0              29.0   
2                   23.0                 40.0              36.0   
3                    NaN                  NaN               NaN   
4                   23.0                 29.0              50.0   

   AKUSSI_spinalpain  KOOSpain  KOOSsymptoms  KOOSdaily_living  KOOSsport  \
0               75.0      67.0          89.0              75.0       70.0   
1               25.0      92.0          96.0              91.0       85.0   
2              100.0      47.0          39.0              31.0        0.0   
3                NaN       NaN           NaN               NaN        NaN   
4              100.0      33.0          29.0              60.0       25.0   

   KOOS_QOL  HAQ_hapVAS  HAQ_haqDI  
0      69.0        63.0       1.50  
1      75.0        23.0       0.25  
2      19.0        53.0       1.75  
3       NaN         NaN        NaN  
4      13.0        70.0       2.00

In [9]:
df2.shape

(219, 17)

In [10]:
# count null values of Protein change allele 1 column
df2['Protein change allele 1 '].isnull().sum()

19

In [11]:
# check if the null value of Protein change allele 1 is the same as Protein change allele 2
df2['Protein change allele 1 '].isnull().sum() == df2['Protein change allele 2'].isnull().sum()

True

In [12]:
# check if tthey are the same indexes
(df2['Protein change allele 1 '].isnull() == df2['Protein change allele 2'].isnull()).all()

False

In [13]:
# print the indexes of the null values
df2[df2['Protein change allele 1 '].isnull()].index

Index([ 50,  54,  61,  66,  72,  80,  84,  87,  88,  89,  90, 165, 166, 183,
       185, 191, 192, 204, 213],
      dtype='int64')

In [14]:
# print the indexes of the null values
null1 = df2['Protein change allele 1 '][df2['Protein change allele 1 '].isnull()].index

In [15]:
null1

Index([ 50,  54,  61,  66,  72,  80,  84,  87,  88,  89,  90, 165, 166, 183,
       185, 191, 192, 204, 213],
      dtype='int64')

In [16]:
null2 = df2['Protein change allele 2'][df2['Protein change allele 2'].isnull()].index

In [17]:
null2

Index([ 50,  54,  61,  66,  80,  84,  87,  88,  89,  90, 165, 166, 178, 183,
       185, 191, 192, 204, 213],
      dtype='int64')

In [18]:
# merge null1 and null2 in a list without repetition
null = list(set(null1) | set(null2))
len(null)

20

In [19]:
len(null1), len(null2)

(19, 19)

In [20]:
# remove the null indexed rows from the dataframe
df2 = df2.drop(null)

In [21]:
df2.shape

(199, 17)

In [22]:
df2.head()

Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
0                    G161R            c.481G>A                  ex8   
1                    G161R            c.481G>A                  ex8   
2                    G161R            c.481G>A                  ex8   
3      Y6_G29del ivs1-1G>A           c.16-1G>A                  in1   
4                    G161R            c.481G>A                  ex8   

  Protein change allele 2 DNA change allele 2 ex/in allele 2  \
0             His371Profs          c.1111dupC           ex13   
1                   G161R            c.481G>A            ex8   
2                   M368V           c.1102A>G           ex13   
3     Y6_G29del ivs1-1G>A           c.16-1G>A            in1   
4                   G161R            c.481G>A            ex8   

   physical_health_score  mental_health_score  AKUSSI_jointpain  \
0                   26.0                 43.0              57.0   
1                   53.0                 51.0              29.0   
2                   23.0                 40.0              36.0   
3                    NaN                  NaN               NaN   
4                   23.0                 29.0              50.0   

   AKUSSI_spinalpain  KOOSpain  KOOSsymptoms  KOOSdaily_living  KOOSsport  \
0               75.0      67.0          89.0              75.0       70.0   
1               25.0      92.0          96.0              91.0       85.0   
2              100.0      47.0          39.0              31.0        0.0   
3                NaN       NaN           NaN               NaN        NaN   
4              100.0      33.0          29.0              60.0       25.0   

   KOOS_QOL  HAQ_hapVAS  HAQ_haqDI  
0      69.0        63.0       1.50  
1      75.0        23.0       0.25  
2      19.0        53.0       1.75  
3       NaN         NaN        NaN  
4      13.0        70.0       2.00

In [23]:
# count the number of 'ex8' value in Exon/intron allele 1 column
df2['Exon/intron allele 1'].value_counts()

Exon/intron allele 1
ex8      55
ex13     21
ex6      18
ex7      14
ex10     12
ex14      8
ex3       8
ex2       7
ex11      7
 in1      6
 ex3      6
ex9       4
 ex13     4
ex12      4
 ex14     3
ex7i      3
ex5       2
in1       2
 ex10     2
in7       2
ex4       2
in8       1
 ex2      1
S47L      1
 ex8      1
in12      1
in5       1
in10      1
in2       1
ex 8      1
Name: count, dtype: int64

vediamo che si deve stare attenti: ci sono varie istanze di ex8, questo perche è stato scritto con gli spazi a volte (ex8, ex8,ex 8)

In [24]:
# sum all the values that contains "ex" in the string value
df2['Exon/intron allele 1'].str.contains('ex').sum()

183

In [25]:
# get the indexes of the values that contains "in" in the string value
in_indexes = df2['Exon/intron allele 1'][df2['Exon/intron allele 1'].str.contains('in')].index

In [26]:
in_indexes

Index([3, 24, 74, 103, 115, 135, 141, 142, 148, 157, 159, 163, 170, 193, 217], dtype='int64')

In [27]:
# drop the rows which index is in_indexes
df3 = df2.drop(in_indexes)

In [28]:
df3.shape

(184, 17)

In [29]:
# sum all the values that contains "ex" in the string value
df3['Exon/intron allele 1'].str.contains('ex').sum()

183

notiamo che 183 su 184 istanze hanno ex nella colonna `Exon/intron allele 1`, vediamo chi è la riga che non ha ex

In [30]:
# see instance that has no "ex" in the string value
df3['Exon/intron allele 1'][~df3['Exon/intron allele 1'].str.contains('ex')]

51    S47L
Name: Exon/intron allele 1, dtype: object

In [31]:
df3[48:49]

Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
51                 c.140C>T                 ex3                 S47L   

   Protein change allele 2 DNA change allele 2 ex/in allele 2  \
51                c.140C>T                 ex3            NaN   

    physical_health_score  mental_health_score  AKUSSI_jointpain  \
51                    NaN                  NaN               NaN   

    AKUSSI_spinalpain  KOOSpain  KOOSsymptoms  KOOSdaily_living  KOOSsport  \
51                NaN       NaN           NaN               NaN        NaN   

    KOOS_QOL  HAQ_hapVAS  HAQ_haqDI  
51       NaN         NaN        NaN

è evidente che i dati sono utilizzabili, ma vanno spostati nelle colonne corrette

siccome vediamo che ci sono altri pazienti con la stessa situazione, non serve nemmeno che facciamo questi cambiamenti perche il paziente 51 e gia rappresentato

In [40]:
# define df that has S47L in the Protein change allele 1 column
df_s = df3[df3['Protein change allele 1 '].str.contains('S47L')]
df_s

Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
6                      S47L            c.140C>T                  ex3   
48                     S47L            c.140C>T                  ex3   
49                     S47L            c.140C>T                  ex3   

   Protein change allele 2 DNA change allele 2 ex/in allele 2  \
6                     S47L            c.140C>T            ex3   
48                    S47L            c.140C>T            ex3   
49                    S47L            c.140C>T            ex3   

    physical_health_score  mental_health_score  AKUSSI_jointpain  \
6                    38.0                 53.0              29.0   
48                   40.0                 49.0              14.0   
49                   28.0                 52.0              71.0   

    AKUSSI_spinalpain  KOOSpain  KOOSsymptoms  KOOSdaily_living  KOOSsport  \
6               100.0      78.0          93.0              71.0       25.0   
48               25.0      89.0          86.0              97.0       75.0   
49               75.0      64.0          64.0              68.0       38.0   

    KOOS_QOL  HAQ_hapVAS  HAQ_haqDI  
6       44.0        50.0       0.75  
48      75.0        20.0       1.50  
49      13.0        80.0       0.50

In [41]:
# get the index of the elements that does not contain "ex" in the string value
not_ex_indexes = df3['Exon/intron allele 1'][~df3['Exon/intron allele 1'].str.contains('ex')].index
not_ex_indexes

Index([51], dtype='int64')

In [42]:
# drop the rows which index is not_ex_indexes
df3 = df3.drop(not_ex_indexes)

In [43]:
not_ex_indexes2 = df3['ex/in allele 2'][~df3['ex/in allele 2'].str.contains('ex')].index
not_ex_indexes2

Index([8, 15, 45, 78, 98, 108, 130, 131, 146, 147, 153, 181, 207], dtype='int64')

In [44]:
# drop the rows which index is not_ex_indexes
df3 = df3.drop(not_ex_indexes2)
df3.shape

(170, 17)

In [45]:
def is_replace(mutation):

    if mutation[0].isupper() and mutation[-1].isupper() and mutation[1:-1].isdigit(): 
        return True
    return False

In [46]:
# drop the rows that does not contain a replacement mutation
df4 = df3[df3['Protein change allele 1 '].apply(is_replace)]
df4.shape

(135, 17)

In [47]:
# in the same way, drop the rows that does not contain a replacement mutation wrt allele 2
df4 = df4[df4['Protein change allele 2'].apply(is_replace)]

In [48]:
df4.shape

(116, 17)

In [50]:
df4.head(15)

Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
1                     G161R            c.481G>A                  ex8   
2                     G161R            c.481G>A                  ex8   
4                     G161R            c.481G>A                  ex8   
5                     G161R            c.481G>A                  ex8   
6                      S47L            c.140C>T                  ex3   
7                     G161R            c.481G>A                  ex8   
10                     W97C            c.291G>C                  ex5   
13                    R225P            c.674G>C                 ex10   
16                     R53Q            c.158G>A                  ex3   
17                    G161R            c.481G>A                  ex8   
18                    G161R            c.481G>A                  ex8   
19                    A267V            c.800C>T                 ex11   
21                    M368V           c.1102A>G                 ex13   
27                    G362E           c.1085G>A                 ex13   
28                    G161R            c.481G>A                  ex8   

   Protein change allele 2 DNA change allele 2 ex/in allele 2  \
1                    G161R            c.481G>A            ex8   
2                    M368V           c.1102A>G           ex13   
4                    G161R            c.481G>A            ex8   
5                    G161R            c.481G>A            ex8   
6                     S47L            c.140C>T            ex3   
7                    G161R            c.481G>A            ex8   
10                    W97C            c.291G>C            ex5   
13                   I216T            c.647T>C            ex9   
16                    R53Q            c.158G>A            ex3   
17                   G161R            c.481G>A            ex8   
18                   G161R            c.481G>A            ex8   
19                   A267V            c.800C>T           ex11   
21                   M368V           c.1102A>G           ex13   
27                   M368V           c.1102A>G           ex13   
28                    R53Q            c.158G>A            ex3   

    physical_health_score  mental_health_score  AKUSSI_jointpain  \
1                    53.0                 51.0              29.0   
2                    23.0                 40.0              36.0   
4                    23.0                 29.0              50.0   
5                    26.0                 31.0              50.0   
6                    38.0                 53.0              29.0   
7                    41.0                 64.0              21.0   
10                   44.0                 56.0              57.0   
13                   33.0                 17.0              57.0   
16                   25.0                 44.0              29.0   
17                   40.0                  2.0              50.0   
18                   16.0                 30.0              43.0   
19                   57.0                 58.0               0.0   
21                   15.0                 50.0              79.0   
27                   38.0                 28.0              29.0   
28                   54.0                 52.0               7.0   

    AKUSSI_spinalpain  KOOSpain  KOOSsymptoms  KOOSdaily_living  KOOSsport  \
1                25.0      92.0          96.0              91.0       85.0   
2               100.0      47.0          39.0              31.0        0.0   
4               100.0      33.0          29.0              60.0       25.0   
5               100.0      44.0          29.0              32.0       35.0   
6               100.0      78.0          93.0              71.0       25.0   
7                 0.0      78.0          39.0              76.0       60.0   
10               50.0      94.0         100.0             100.0      100.0   
13              100.0      17.0           0.0              21.0        0.0   
16               7

In [52]:
seq = [
    'Protein change allele 1 ', 
    'DNA change allele 1',
    'Exon/intron allele 1', 
    'Protein change allele 2',
    'DNA change allele 2', 
    'ex/in allele 2'
]

# keep only the columns in the seq list
df5 = df4[seq]
df5.head()

Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
1                    G161R            c.481G>A                  ex8   
2                    G161R            c.481G>A                  ex8   
4                    G161R            c.481G>A                  ex8   
5                    G161R            c.481G>A                  ex8   
6                     S47L            c.140C>T                  ex3   

  Protein change allele 2 DNA change allele 2 ex/in allele 2  
1                   G161R            c.481G>A            ex8  
2                   M368V           c.1102A>G           ex13  
4                   G161R            c.481G>A            ex8  
5                   G161R            c.481G>A            ex8  
6                    S47L            c.140C>T            ex3

In [54]:
df5['DNA change allele 1'].value_counts()

DNA change allele 1
c.481G>A      36
c.365C>T      12
 c.1201G>C     8
c.1102A>G      6
 c.158G>A      4
 c.1102A>G     3
 C158G>A       3
c.502G>A       3
c.140C>T       3
 c.688C>T      3
c.688C>T       3
c.647T>C       2
c.368G>C       2
c.1078G>C      2
c.808G>A       2
c.217T>C       1
c.289T>G       1
c.509G>C       1
c.1081G>A      1
c.553G>A       1
c.449C>T       1
c.815A>G       1
c.508G>A       1
c.359G>T       1
c.130C>T       1
c.1079G>C      1
c.680T>C       1
c.593G>A       1
c.742A>G       1
c.119A>C       1
c.752G>A       1
c.347T>C       1
c.533A>G       1
c.990G>C       1
c.1085G>A      1
c.800C>T       1
c.674G>C       1
c.291G>C       1
c.1057A>C      1
Name: count, dtype: int64

some instances have a space before the DNA change: let's uniform the data

In [57]:
# eliminate the space in the column DNA change allele 1
df5['DNA change allele 1'] = df5['DNA change allele 1'].str.replace(' ', '')
df5['DNA change allele 1'].value_counts()

/var/folders/xd/64006ztx62g6gm6fh5jj4src0000gn/T/ipykernel_26826/2035421048.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df5['DNA change allele 1'] = df5['DNA change allele 1'].str.replace(' ', '')


DNA change allele 1
c.481G>A     36
c.365C>T     12
c.1102A>G     9
c.1201G>C     8
c.688C>T      6
c.158G>A      4
C158G>A       3
c.502G>A      3
c.140C>T      3
c.1078G>C     2
c.368G>C      2
c.647T>C      2
c.808G>A      2
c.289T>G      1
c.359G>T      1
c.217T>C      1
c.509G>C      1
c.1081G>A     1
c.553G>A      1
c.1079G>C     1
c.815A>G      1
c.508G>A      1
c.449C>T      1
c.742A>G      1
c.680T>C      1
c.593G>A      1
c.130C>T      1
c.119A>C      1
c.752G>A      1
c.347T>C      1
c.533A>G      1
c.990G>C      1
c.1085G>A     1
c.800C>T      1
c.674G>C      1
c.291G>C      1
c.1057A>C     1
Name: count, dtype: int64

In [58]:
# see value counts for Exon/intron allele 1 column
df5['Exon/intron allele 1'].value_counts()

Exon/intron allele 1
ex8      41
ex6      16
ex13     12
ex10      8
ex3       6
 ex3      6
ex14      5
ex11      4
ex9       4
 ex14     3
 ex13     3
ex5       2
 ex10     2
ex12      1
ex7       1
ex4       1
ex 8      1
Name: count, dtype: int64

In [59]:
# eliminate the space in the column Exon/intron allele 1
df5['Exon/intron allele 1'] = df5['Exon/intron allele 1'].str.replace(' ', '')
df5['Exon/intron allele 1'].value_counts()

/var/folders/xd/64006ztx62g6gm6fh5jj4src0000gn/T/ipykernel_26826/4246110940.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df5['Exon/intron allele 1'] = df5['Exon/intron allele 1'].str.replace(' ', '')


Exon/intron allele 1
ex8     42
ex6     16
ex13    15
ex3     12
ex10    10
ex14     8
ex11     4
ex9      4
ex5      2
ex12     1
ex7      1
ex4      1
Name: count, dtype: int64

In [61]:
df5['DNA change allele 2'].value_counts()

DNA change allele 2
c.481G>A      22
c.365C>T      13
 c.1201G>C     9
c.1102A>G      8
c.1078G>C      6
 c.158G>A      6
 c.688C>T      4
c.808G>A       4
 c.899T>G      4
c.680T>C       3
 C158G>A       3
 c.1102A>G     3
c.140C>T       3
c.502G>A       2
c.688C>T       2
c.647T>C       2
c.368G>C       2
c.533A>G       1
c.899T>G       1
c.1057A>C      1
c.815A>G       1
c.995C>G       1
c.500C>T       1
c.359G>T       1
c.119A>C       1
c.593G>A       1
c.130C>T       1
c.742A>G       1
c.752G>A       1
c.614G>A       1
c.347T>C       1
c.990G>C       1
c.1037T>C      1
c.454G>A       1
c.800C>T       1
c.291G>C       1
c.52G>T        1
Name: count, dtype: int64

In [62]:
# eliminate the space in the column DNA change allele 2
df5['DNA change allele 2'] = df5['DNA change allele 2'].str.replace(' ', '')
df5['DNA change allele 2'].value_counts()

/var/folders/xd/64006ztx62g6gm6fh5jj4src0000gn/T/ipykernel_26826/1093346150.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df5['DNA change allele 2'] = df5['DNA change allele 2'].str.replace(' ', '')


DNA change allele 2
c.481G>A     22
c.365C>T     13
c.1102A>G    11
c.1201G>C     9
c.688C>T      6
c.1078G>C     6
c.158G>A      6
c.899T>G      5
c.808G>A      4
C158G>A       3
c.140C>T      3
c.680T>C      3
c.502G>A      2
c.647T>C      2
c.368G>C      2
c.593G>A      1
c.995C>G      1
c.500C>T      1
c.815A>G      1
c.359G>T      1
c.533A>G      1
c.1057A>C     1
c.752G>A      1
c.130C>T      1
c.742A>G      1
c.119A>C      1
c.614G>A      1
c.347T>C      1
c.990G>C      1
c.1037T>C     1
c.454G>A      1
c.800C>T      1
c.291G>C      1
c.52G>T       1
Name: count, dtype: int64

In [63]:
df5['ex/in allele 2'].value_counts()

ex/in allele 2
ex8      25
ex6      17
ex13     16
 ex3     10
 ex14     9
ex10      7
ex11      6
ex3       4
ex9       4
 ex12     4
 ex10     4
ex12      3
 ex13     3
ex5       1
ex7       1
ex2       1
ex 8      1
Name: count, dtype: int64

In [64]:
# eliminate the space in the column ex/in allele 2
df5['ex/in allele 2'] = df5['ex/in allele 2'].str.replace(' ', '')

/var/folders/xd/64006ztx62g6gm6fh5jj4src0000gn/T/ipykernel_26826/4150708269.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df5['ex/in allele 2'] = df5['ex/in allele 2'].str.replace(' ', '')


In [65]:
df5['ex/in allele 2'].value_counts()

ex/in allele 2
ex8     26
ex13    19
ex6     17
ex3     14
ex10    11
ex14     9
ex12     7
ex11     6
ex9      4
ex5      1
ex7      1
ex2      1
Name: count, dtype: int64

In [70]:
df5.head()

Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
1                    G161R            c.481G>A                  ex8   
2                    G161R            c.481G>A                  ex8   
4                    G161R            c.481G>A                  ex8   
5                    G161R            c.481G>A                  ex8   
6                     S47L            c.140C>T                  ex3   

  Protein change allele 2 DNA change allele 2 ex/in allele 2  
1                   G161R            c.481G>A            ex8  
2                   M368V           c.1102A>G           ex13  
4                   G161R            c.481G>A            ex8  
5                   G161R            c.481G>A            ex8  
6                    S47L            c.140C>T            ex3

notiamo che il paziente 4 e 5 sono uguali all'1 in tutto e per tutto, quindi saranno rappresentati dalla proteina 3d ottenuta per il paziente1

In [68]:
# count how many different rows are in the dataframe
df5.drop_duplicates().shape

(52, 6)

In [69]:
df6 = df5.drop_duplicates()
df6.head()

Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
1                     G161R            c.481G>A                  ex8   
2                     G161R            c.481G>A                  ex8   
6                      S47L            c.140C>T                  ex3   
10                     W97C            c.291G>C                  ex5   
13                    R225P            c.674G>C                 ex10   

   Protein change allele 2 DNA change allele 2 ex/in allele 2  
1                    G161R            c.481G>A            ex8  
2                    M368V           c.1102A>G           ex13  
6                     S47L            c.140C>T            ex3  
10                    W97C            c.291G>C            ex5  
13                   I216T            c.647T>C            ex9

vediamo infatti che i pazienti 4 e 5 sono stati tolti

In [ ]:
#------------------------------------------------------------

In [133]:
# return the indexes of the elements that have the same value in both columns 
same_change = df4[df4['Protein change allele 1 '] == df4['Protein change allele 2']].index
print(len(same_change))
# do the same for exon/intron columns
same_exon = df4[df4['Exon/intron allele 1'] == df4['ex/in allele 2']].index
print(len(same_exon))

# count how many indexes are in both lists
indexes = set(same_change) & set(same_exon)

# print the df4 elements that are in the indexes list
df4.loc[list(indexes)]


78
76


Protein change allele 1  DNA change allele 1 Exon/intron allele 1  \
128                    A122V            c.365C>T                  ex6   
1                      G161R            c.481G>A                  ex8   
129                    A122V            c.365C>T                  ex6   
4                      G161R            c.481G>A                  ex8   
5                      G161R            c.481G>A                  ex8   
..                       ...                 ...                  ...   
111                     R53Q            c.158G>A                  ex3   
112                    P230S            c.688C>T                 ex10   
114                    G123A            c.368G>C                  ex6   
122                    G161R            c.481G>A                  ex8   
127                    A122V            c.365C>T                  ex6   

    Protein change allele 2 DNA change allele 2 ex/in allele 2  \
128                   A122V            c.365C>T            ex6   
1                     G161R            c.481G>A            ex8   
129                   A122V            c.365C>T            ex6   
4                     G161R            c.481G>A            ex8   
5                     G161R            c.481G>A            ex8   
..                      ...                 ...            ...   
111                    R53Q            c.158G>A            ex3   
112                   P230S            c.688C>T           ex10   
114                   G123A            c.368G>C            ex6   
122                   G161R            c.481G>A            ex8   
127                   A122V            c.365C>T            ex6   

     physical_health_score  mental_health_score  AKUSSI_jointpain  \
128                   24.0                 50.0              29.0   
1                     53.0                 51.0              29.0   
129                   40.0                 60.0              14.0   
4                     23.0                 29.0              50.0   
5                     26.0                 31.0              50.0   
..                     ...                  ...               ...   
111                   37.0                 40.0              21.0   
112                   39.0                 32.0              50.0   
114                   29.0                 37.0              71.0   
122                    NaN                  NaN              14.0   
127                   28.0                 47.0              29.0   

     AKUSSI_spinalpain  KOOSpain  KOOSsymptoms  KOOSdaily_living  KOOSsport  \
128               25.0      44.0          61.0              25.0       20.0   
1                 25.0      92.0          96.0              91.0       85.0   
129               25.0      78.0          86.0              75.0       85.0   
4                100.0      33.0          29.0              60.0       25.0   
5                100.0      44.0          29.0              32.0       35.0   
..                 ...       ...           ...               ...        ...   
111               50.0      83.0          93.0              87.0       80.0   
112               75.0      64.0          54.0              65.0        5.0   
114               50.0      44.0          50.0              52.0       40.0   
122                0.0     100.0         100.0             100.0      100.0   
127               25.0      53.0          82.0              52.0        NaN   

     KOOS_QOL  HAQ_hapVAS  HAQ_haqDI  
128      50.0       100.0       1.88  
1        75.0        23.0       0.25  
129      63.0         0.0       0.00  
4        13.0        70.0       2.00  
5        13.0        71.0       1.88  
..        ...         ...        ...  
111      75.0        75.0       0.88  
112      19.0        30.0       1.13  
114      56.0        35.0       1.75  
122     100.0        30.0       0.13  
127      63.0        50.0       1.13  

[71 rows x 17 columns]

In [134]:
# count unique values of Protein change allele 1 column
df4['Protein change allele 1 '].nunique()

38